contracts/facets/EmbalmerFacet.sol

```diff
@@ -46,11 +62,27 @@ contract EmbalmerFacet {
             revert LibErrors.ResurrectionTimeInPast(resurrectionTime);
         }
 
-        // Initialize a list of archaeologist addresses
+        // Confirm that archaeologists are provided
+        if (archaeologists.length == 0) {
+            revert LibErrors.NoArchaeologistsProvided();
+        }
+
+        // Initialize a list of archaeologist addresses to be passed in to the
+        // sarcophagus object
         address[] memory archaeologistAddresses = new address[](
             archaeologists.length
         );
 
+        // Initialize the storage fee of the archaeologist who uploades to
+        // arweave. This will be obtained in the for loop and stored on the
+        // sarcophagus object.
+        uint256 storageFee = 0;
+
+        // We need to iterate over every archaeologist to
+        //   - lock up archaeologist's free bond
+        //   - get the storage fee from the arweave archaeologist
+        //   - store each archaeologist's bounty, digging fee, and unencrypted shard in app storage
+        //   - get the archaeologist's address to store on the sarcophagus
```

Comment:
I missed this during previous review -- I recall us discussing locking up the arch's cursed bond amounts during the `finalize` transaction (instead of in `initialize`)

This was b/c once those cursed bonds are locked, they can only be returned if the embalmer cancels the sarcophagus. If this is moved to `finalize`, then the archaeologists can guarantee the sarcophagus is "live" if those cursed bonds are locked.

The alternative would be to allow archaeologists to call `cancel` on a sarcophagus after some time period (maybe 24 hrs), if we wanted to leave the locking up of bonds here.

Reply:
Yes, I remember now. It makes more sense to have this in finalize.